frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditFieldDrawer/edit-fieldtype/EditAttachment/EditAttachment.tsx

```diff
@@ -0,0 +1,197 @@
+import {
+  AttachmentFieldBase,
+  AttachmentSize,
+  BasicField,
+  FormFieldDto,
+} from '~shared/types/field'
+
+import { createBaseValidationRules } from '~utils/fieldValidation'
+import { SingleSelect } from '~components/Dropdown'
+import type { ComboboxItem } from '~components/Dropdown/types'
+import FormErrorMessage from '~components/FormControl/FormErrorMessage'
+import FormLabel from '~components/FormControl/FormLabel'
+import InlineMessage from '~components/InlineMessage'
+import Input from '~components/Input'
+import Textarea from '~components/Textarea'
+import Toggle from '~components/Toggle'
+
+import { useCreateTabForm } from '~features/admin-form/create/builder-and-design/useCreateTabForm'
+
+import { DrawerContentContainer } from '../common/DrawerContentContainer'
+import { FormFieldDrawerActions } from '../common/FormFieldDrawerActions'
+import { EditFieldProps } from '../common/types'
+import { useEditFieldForm } from '../common/useEditFieldForm'
+
+import { AttachmentStackedBar } from './AttachmentStackedBar'
+
+type EditAttachmentProps = EditFieldProps<AttachmentFieldBase>
+
+const EDIT_ATTACHMENT_FIELD_KEYS = [
+  'title',
+  'description',
+  'required',
+  'attachmentSize',
+] as const
+
+type EditAttachmentKeys = typeof EDIT_ATTACHMENT_FIELD_KEYS[number]
+
+type EditAttachmentInputs = Pick<AttachmentFieldBase, EditAttachmentKeys>
+
+const transformAttachmentFieldToEditForm = (
+  field: AttachmentFieldBase,
+): EditAttachmentInputs => {
+  return pick(field, EDIT_ATTACHMENT_FIELD_KEYS)
+}
+
+const transformAttachmentEditFormToField = (
+  inputs: EditAttachmentInputs,
+  originalField: AttachmentFieldBase,
+): AttachmentFieldBase => {
+  return extend({}, originalField, inputs)
+}
+
+export const EditAttachment = ({ field }: EditAttachmentProps): JSX.Element => {
+  const { data: form } = useCreateTabForm()
+  const {
+    register,
+    formState: { errors },
+    control,
+    getValues,
+    isSaveEnabled,
+    buttonText,
+    handleUpdateField,
+    isLoading,
+    handleCancel,
+  } = useEditFieldForm<EditAttachmentInputs, AttachmentFieldBase>({
+    field,
+    transform: {
+      input: transformAttachmentFieldToEditForm,
+      output: transformAttachmentEditFormToField,
+    },
+  })
+
+  const requiredValidationRule = useMemo(
+    () => createBaseValidationRules({ required: true }),
+    [],
+  )
+
+  const otherAttachmentsSize = useMemo(() => {
+    if (!form?.form_fields) return 0
+    return form.form_fields
+      .filter(
+        (ff): ff is FormFieldDto<AttachmentFieldBase> =>
+          ff.fieldType === BasicField.Attachment && ff._id !== field._id,
+      )
+      .reduce((sum, ff) => sum + Number(ff.attachmentSize), 0)
+  }, [field._id, form?.form_fields])
+
+  const maxTotalSizeMb: number = useMemo(() => {
+    if (!form?.responseMode) return 0
+    switch (form.responseMode) {
+      case FormResponseMode.Email:
+        return 7
+      case FormResponseMode.Encrypt:
+        return 20
```

Comment:
Should we not use the constants here? 🤔 
```suggestion
        return Number(AttachmentSize.SevenMb)
      case FormResponseMode.Encrypt:
        return Number(AttachmentSize.TwentyMb)
```

Reply:
849f756